core/src/main/java/hudson/model/Executor.java

```diff
@@ -355,6 +355,10 @@ public SubTask call() throws Exception {
                         LOGGER.log(FINE, getName()+" grabbed "+workUnit+" from queue");
                     SubTask task = workUnit.work;
                     Executable executable = task.createExecutable();
+                    if (executable == null) {
+                        String displayName = task instanceof Queue.Task ? ((Queue.Task) task).getFullDisplayName() : task.getDisplayName();
+                        LOGGER.log(WARNING, "{0} cannot be run (for example because it is disabled)", displayName);
+                    }
```

Comment:
I thought that was exactly what we did not want to do, based on your own earlier comment:

> In such case Jenkins starts the workUnit in order to report results to console outputs.